src/core_modules/capture-core/components/Pages/Enrollment/EnrollmentPage.container.js

```diff
@@ -104,6 +104,7 @@ export const EnrollmentPage: ComponentType<{||}> = () => {
     },
     [
         dispatch,
+        teiId,
```

Comment:
I did sneak in another fix here: When changing tei id from the url the page would not correctly update. Seems like this did the trick.